### ISSUE TITLE
VS crashes while editing a class constructor with errors - back and forth to get the typing right

### ISSUE THREAD

**Issue by jinglouMSFT** [microsoft/poshtools#349]
Powershell v5 from Feb 2015 preview. 

Type the following:
class Beer {
     #size of beer in ounces
     [uint32] $Size;

```
 Beer([int] @)  <==== I made few mistakes when I was typing this, so the cursor was moving back and forth before I got it right.   
```

Result:  Exception.
Here is the callstack:

`System.ArgumentNullException: Value cannot be null.&#x000D;&#x000A;Parameter name: classification&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Classification.ClassificationSpan..ctor(SnapshotSpan span, IClassificationType classification)&#x000D;&#x000A;   at PowerShellTools.Classification.PowerShellClassifier.AddTokenClassifications(ITextBuffer buffer, SnapshotSpan span, List`1 classifications, Nullable`1 lastClassificationSpan, IClassificationType gapType)&#x000D;&#x000A;   at PowerShellTools.Classification.PowerShellClassifier.VirtualGetClassificationSpans(SnapshotSpan span)&#x000D;&#x000A;   at PowerShellTools.Classification.Classifier.GetClassificationSpans(SnapshotSpan span)&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Classification.Implementation.ClassifierTagger.&lt;GetTags&gt;d__1.MoveNext()&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.&lt;GetTagsForBuffer&gt;d__1.MoveNext()`

**Comment by ASayre** [comment]
Also getting it:

`
  <entry>
    <record>1522</record>
    <time>2015/04/02 23:45:27.902</time>
    <type>Error</type>
    <source>Editor or Editor Extension</source>
    <description>System.ArgumentNullException: Value cannot be null.&#x000D;&#x000A;Parameter name: tag&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Tagging.TagSpan`1..ctor(SnapshotSpan span, T tag)&#x000D;&#x000A;   at PowerShellTools.Classification.TagInformation`1.GetTagSpan(ITextSnapshot snapshot)&#x000D;&#x000A;   at PowerShellTools.Classification.PowerShellOutliningTagger.&lt;&gt;c__DisplayClass4.&lt;GetTags&gt;b__0(TagInformation`1 current)&#x000D;&#x000A;   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()&#x000D;&#x000A;   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()&#x000D;&#x000A;   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)&#x000D;&#x000A;   at PowerShellTools.Classification.PowerShellOutliningTagger.GetTags(NormalizedSnapshotSpanCollection spans)&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.&lt;GetTagsForBuffer&gt;d__1.MoveNext()</description>
  </entry>
`

I used @ChrisLynchHP 's https://raw.githubusercontent.com/HewlettPackard/POSH-HPOneView/1.20/HPOneView.120.psm1 as a test file, typed a whole bunch of stuff, and was navigating around.

**Comment by HuanhuanSunMSFT** [comment]
This should be fixed in the PR fixing race condition during colorization and tagging.
@AndreSayreMSFT @jinglou-

**Comment by ASayre** [comment]
Agreed.  Wasn't able to get this to break/repro again; as it was fixed with PR #338
